### PR TITLE
KAFKA-17543: Improve and clarify the error message about generated broker IDs in migration

### PR DESCRIPTION
**Pull request by scholzj**
This PR tries to improve the error message when `broker.id` is set to `-1` when migration is enabled. It is not needed to disable the `broker.id.generation.enable` option. It is sufficient to just not use it (by not setting the `broker.id` to `-1`). This was discussed in 

### Committer Checklist (excluded from commit message)

- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: core/src/main/scala/kafka/server/KafkaConfig.scala [hunk 1]
```diff
@@ -967,7 +967,7 @@ class KafkaConfig private(doLog: Boolean, val props: util.Map[_, _])
       // ZK-based
       if (migrationEnabled) {
         require(brokerId >= 0,
-          "broker broker.id.generation.enable is incompatible with migration to ZK. Please disable it before enabling migration")
+          "broker.id generation is incompatible with ZooKeeper migration. Please stop using it before enabling migration (set broker.id to a value greater or equal to 0).")
```
Comment (chia7712) [inline core/src/main/scala/kafka/server/KafkaConfig.scala:970]: this path is unrelated to broker.id generation, right?

**Review by chia7712 — COMMENTED** [review]
@scholzj thanks for your patch

File: core/src/main/scala/kafka/server/KafkaConfig.scala (hunk 1 above)
Comment (scholzj) [reply core/src/main/scala/kafka/server/KafkaConfig.scala:970]: I'm not entirely sure why it is here -> I just updated the message. The test does not trigger it. But to be honest, my Kafka code knowledge is not good enough to understand if under some circumstances it might get here.

Comment (showuon) [reply core/src/main/scala/kafka/server/KafkaConfig.scala:970]: This path and the above one (L850) are both under `validateValues` method. The above one is validate for ZK brokers (i.e. process.role is null). And this line is to validate KRaft, or migrating brokers. Both are related to this fix.

File: core/src/main/scala/kafka/server/KafkaConfig.scala [hunk 2]
```diff
@@ -847,7 +847,7 @@ class KafkaConfig private(doLog: Boolean, val props: util.Map[_, _])
       }
       if (brokerIdGenerationEnable) {
         if (migrationEnabled) {
-          require(brokerId != -1, "broker id generation is incompatible with migration to ZK. Please disable it before enabling migration")
+          require(brokerId != -1, "broker.id generation is incompatible with ZooKeeper migration. Please stop using it before enabling migration (set broker.id to a value greater or equal to 0).")
```
Comment (showuon) [inline core/src/main/scala/kafka/server/KafkaConfig.scala]: nit: I think we can set the condition explicitly to `brokerId >= 0` in this line. But of course, it's a pretty minor comment. It's fine we merge it as is for v3.9.0.

**Review by showuon — APPROVED** [review]
LGTM!

File: core/src/main/scala/kafka/server/KafkaConfig.scala (hunk 1 above)
Comment (chia7712) [reply core/src/main/scala/kafka/server/KafkaConfig.scala:970]: Maybe we should just say "please define the broker.id for migration"? It can fail due to undefined broker.if even though we have disabled generation.

File: core/src/main/scala/kafka/server/KafkaConfig.scala (hunk 2 above)
Comment (scholzj) [reply core/src/main/scala/kafka/server/KafkaConfig.scala]: I updated it to `>= 0`. It makes sense to use the same in all checks.

File: core/src/main/scala/kafka/server/KafkaConfig.scala (hunk 1 above)
Comment (scholzj) [reply core/src/main/scala/kafka/server/KafkaConfig.scala:970]: You mean in the error message? Something like `Please define the broker.id for ZooKeeper migration`?

Comment (chia7712) [reply core/src/main/scala/kafka/server/KafkaConfig.scala:970]: never mind. Current version is good enough. I will merge it later

**Review by chia7712 — APPROVED** [review]

**Comment by chia7712** [comment]
3.9 RC is almost there and this PR should be backport to 3.9, hence I will wait for RC 1 ...
